src/containers/NonBitGoRecovery/Forms/EthereumForm.tsx

```diff
@@ -1,103 +1,273 @@
-import { Selectfield, Textarea, Textfield } from '../../../components';
+import * as React from 'react';
+import * as Yup from 'yup';
+import { Formik, Form } from 'formik';
+import {
+  Button,
+  FormikSelectfield,
+  FormikTextarea,
+  FormikTextfield,
+  Icon,
+  Notice,
+} from '../../../components';
+import { useOutletContext, useParams } from 'react-router-dom';
+import { useElectronCommand, useElectronQuery } from '../../../hooks';
+
+const validationSchema = Yup.object({
+  krsProvider: Yup.mixed()
+    .oneOf(['keyternal', 'bitgoKRSv2', 'dai'])
+    .label('Key Recovery Service'),
+  userKey: Yup.string().required(),
+  backupKey: Yup.string().required(),
+  bitgoKey: Yup.string().required(),
+  walletPassphrase: Yup.string().required(),
+  recoveryDestination: Yup.string().required(),
+  scan: Yup.string().required(),
+});
 
 export default function EthereumForm() {
+  const params = useParams<'BitGoEnvironment' | '*'>();
+  const BitGoEnvironment = params.BitGoEnvironment;
+  const coinTicker = BitGoEnvironment === 'prod' ? 'eth' : 'gteth';
+
+  const [, setAlert] =
+    useOutletContext<
+      [
+        string | undefined,
+        React.Dispatch<React.SetStateAction<string | undefined>>
+      ]
+    >();
+
+  const isRecovering = React.useRef(false);
+
+  const [recover, recoverPayload] = useElectronCommand('recover');
+  const [showSaveDialog, showSaveDialogPayload] =
+    useElectronCommand('showSaveDialog');
+  const [writeFile, writeFilePayload] = useElectronCommand('writeFile');
+  const chainPayload = useElectronQuery('getChain', [coinTicker]);
+
+  const isLoading =
+    recoverPayload.state === 'loading' ||
+    showSaveDialogPayload.state === 'loading' ||
+    writeFilePayload.state === 'loading' ||
+    chainPayload.state === 'loading';
+  const recoverButtonLabel = isLoading ? 'Recovering...' : 'Recover Funds';
+
+  React.useEffect(() => {
+    if (recoverPayload.state === 'success') {
+      let recoveryTransaction;
+      if ('txHex' in recoverPayload.data) {
+        recoveryTransaction = recoverPayload.data.txHex;
+      } else if ('transactionHex' in recoverPayload.data) {
+        recoveryTransaction = recoverPayload.data.transactionHex;
+      }
+
+      if (!recoveryTransaction) {
+        setAlert('Fully-signed recovery transaction not detected.');
+        return;
+      }
+
+      const fileName =
+        chainPayload.data + '-recovery-' + Date.now().toString() + '.json';
+      const dialogParams = {
+        filters: [
+          {
+            name: 'Custom File Type',
+            extensions: ['json'],
+          },
+        ],
+        defaultPath: `~/${fileName}`,
+      };
```

Comment:
@benpreiser you're not referencing this value in multiple places, just inline it to the showSaveDialog call